src/providers/krb5/krb5_auth.c

```diff
@@ -291,21 +295,38 @@ static errno_t krb5_auth_prepare_ccache_name(struct krb5child_req *kr,
 {
     const char *ccname_template;
 
-    ccname_template = dp_opt_get_cstring(kr->krb5_ctx->opts, KRB5_CCNAME_TMPL);
+    switch (kr->dom->type) {
+    case DOM_TYPE_POSIX:
+        ccname_template = dp_opt_get_cstring(kr->krb5_ctx->opts, KRB5_CCNAME_TMPL);
 
-    kr->ccname = expand_ccname_template(kr, kr, ccname_template,
-                                        kr->krb5_ctx->illegal_path_re, true,
-                                        be_ctx->domain->case_sensitive);
-    if (kr->ccname == NULL) {
-        DEBUG(SSSDBG_CRIT_FAILURE, "expand_ccname_template failed.\n");
-        return ENOMEM;
-    }
+        kr->ccname = expand_ccname_template(kr, kr, ccname_template,
+                                            kr->krb5_ctx->illegal_path_re, true,
+                                            be_ctx->domain->case_sensitive);
+        if (kr->ccname == NULL) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "expand_ccname_template failed.\n");
+            return ENOMEM;
+        }
 
-    kr->old_ccname = ldb_msg_find_attr_as_string(user_msg,
-                                                 SYSDB_CCACHE_FILE, NULL);
-    if (kr->old_ccname == NULL) {
-        DEBUG(SSSDBG_TRACE_LIBS,
-                "No ccache file for user [%s] found.\n", kr->pd->user);
+        kr->old_ccname = ldb_msg_find_attr_as_string(user_msg,
+                                                    SYSDB_CCACHE_FILE, NULL);
+        if (kr->old_ccname == NULL) {
+            DEBUG(SSSDBG_TRACE_LIBS,
+                    "No ccache file for user [%s] found.\n", kr->pd->user);
+        }
+        break;
+    case DOM_TYPE_APPLICATION:
+        DEBUG(SSSDBG_TRACE_FUNC,
+               "Domain type application, will use in-memory ccache\n");
+        kr->ccname = talloc_asprintf(kr, "MEMORY:%s", kr->pd->user);
```

Comment:
I think you can just use a dummy value here to not confuse the parameter transfer to krb5_child. Since krb5_child knows if to domain is a posix domain or not you can then just skip the creation (and  the removal added below) of the credential cache. Creating the ccache is not needed because authentication is successful if krb5_get_init_creds_password() return 0 (and validate_tgt() as well).